src/ImageBuilder.ts

```diff
@@ -62,13 +62,22 @@ export default class ImageBuilder {
 
             var isCreateBlob = false;
             var imgBuilderId = "";
-            
+
             if (this._taskParameters.customizerSource != undefined && this._taskParameters.customizerSource != "") {
                 isCreateBlob = true;
             }
 
             if (!this._taskParameters.isTemplateJsonProvided) {
-                imgBuilderId = `/subscriptions/${subscriptionId}/resourcegroups/${this._taskParameters.resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/${this.idenityName}`;
+                if (this.idenityName.indexOf("subscriptions/") > -1) {
```

Comment:
I would use startsWith here instead of indexOf. Check if startsWith is available or not.

Reply:
It exists but i have to test again after making this change.Its working fine with indesof.Do i need to make this change

Reply:
done